pysoa/server/server.py

```diff
@@ -300,9 +300,18 @@ def setup(self):
 
     def _close_old_django_connections(self):
         if self.use_django:
-            from django.db import close_old_connections
-            self.logger.debug('Cleaning Django connections')
-            close_old_connections()
+            try:
+                from django.db import transaction
+                if transaction.get_autocommit():
+                    from django.db import close_old_connections
+                    self.logger.debug('Cleaning Django connections')
+                    close_old_connections()
+            except BaseException as e:
```

Comment:
why `BaseException` and not `Exception`?

Reply:
Because PyTest's `Failed` exception extends `BaseException`, not `Exception`.